src/js/domain/LottoMachine.js

```diff
@@ -0,0 +1,88 @@
+import View from "../view/view";
+import { LottoRank } from "./LottoRank";
+import { lottoMoneyRule } from "../rules";
+import { Lotto, WinningLotto } from "./Lotto";
+import { LOTTO_LENGTH, MAXIMUM_LOTTO_NUMBER } from "../constants";
+
+class LottoMachine {
+  static LOTTO_PRICE = 1000;
+
+  lottos = [];
+  constructor() {}
+
+  buy(money) {
+    if (!lottoMoneyRule.validates(money)) return;
+
+    const lottos = new Set();
+    const theNumberOfLottos = this.countTheNumberOfLottos(money);
+
+    while (lottos.size < theNumberOfLottos) {
+      const lotto = this.generateLottoNumbers();
+      lottos.add(JSON.stringify(lotto));
+    }
+
+    this.lottos = Array.from(lottos).map((lotto) => new Lotto(JSON.parse(lotto)));
+
+    View.printLottoInfo(this.lottos);
+    return this.lottos;
+  }
+
+  /**
+   * @param {WinningLotto} winningLotto
+   * @returns {Map}
+   */
+  getLottoRanks(winningLotto) {
+    const lottoRanks = this.lottos.map((lotto) => winningLotto.getRank(lotto));
+    const lottoRankCounts = this.countLottoRanks(lottoRanks);
+    return lottoRankCounts;
+  }
+
+  /**
+   * @param {string[]} lottoRanks
+   * @returns {Map}
+   */
+  countLottoRanks(lottoRanks) {
+    return lottoRanks.reduce((acc, rank) => {
+      acc.set(rank, acc.get(rank) + 1 || 1);
+      return acc;
+    }, new Map());
+  }
+
+  countTheNumberOfLottos(money) {
+    return money / LottoMachine.LOTTO_PRICE;
+  }
+
+  generateLottoNumbers() {
+    const lottos = new Set();
+
+    while (lottos.size < LOTTO_LENGTH) {
+      const randomNumber = this.generateRandomNumbers();
+      lottos.add(randomNumber);
+    }
+    return Array.from(lottos).sort((a, b) => a - b);
+  }
```

Comment:
```suggestion
  generateLotto() {
    const lotto = new Set();

    while (lotto.size < LOTTO_LENGTH) {
      const randomNumber = this.generateRandomNumbers();
      lotto.add(randomNumber);
    }
    return Array.from(lotto).sort((a, b) => a - b);
  }
```
이렇게 해야 자연스럽지 않을까 싶어요..!

그리고 로또 객체를 만들 때 정렬해서 저장해주면 되지 않을까요~?

Reply:
어떠한 경우에도 6번의 반복으로 중복되지 않은 숫자를 만들기 위해선 어떻게 해야 좋을까요?
힌트를 조금 드리자면, 랜덤으로 값을 생성하지 않으면 된답니다!